app/questionnaire/rules.py

```diff
@@ -294,6 +294,9 @@ def get_answer_store_value(answer_id, answer_store, schema, group_instance, grou
         raise Exception('Multiple answers ({:d}) found evaluating when rule for answer ({})'
                         .format(filtered.count(), answer_id))
 
+    if filtered.answers and isinstance(filtered.answers[0]['value'], dict):
+        return filtered.answers[0]['value']['years']
```

Comment:
Not sure how/what way to make this more generic.. Thought about something like below
```
    if filtered.answers and isinstance(filtered.answers[0]['value'], dict):
        for value in filtered.answers[0]['value'].values():
            return value
```